R/io-csv.R

```diff
@@ -0,0 +1,282 @@
+#' their last access time, in seconds. Uses the `POLARS_FILE_CACHE_TTL`
+#' environment variable (which defaults to 1 hour) if not given.
+#' @param include_file_paths Include the path of the source file(s) as a column
+#' with this name.
+#'
+#' @inherit as_polars_lf return
+#' @examples
+#' my_file <- tempfile()
+#' write.csv(iris, my_file)
+#' lazy_frame <- pl$scan_csv(my_file)
+#' lazy_frame$collect()
+#' unlink(my_file)
+pl__scan_csv <- function(
+    source,
+    ...,
+    has_header = TRUE,
+    separator = ",",
+    comment_prefix = NULL,
+    quote_char = '"',
+    skip_rows = 0,
+    schema = NULL,
+    schema_overrides = NULL,
+    null_values = NULL,
+    missing_utf8_is_empty_string = FALSE,
+    ignore_errors = FALSE,
+    cache = FALSE,
+    # with_column_names = NULL,
+    infer_schema = TRUE,
+    infer_schema_length = 100,
+    n_rows = NULL,
+    encoding = c("utf8", "utf8-lossy"),
+    low_memory = FALSE,
+    rechunk = FALSE,
+    skip_rows_after_header = 0,
+    row_index_name = NULL,
+    row_index_offset = 0,
+    try_parse_dates = FALSE,
+    eol_char = "\n",
+    # new_columns = NULL,
+    raise_if_empty = TRUE,
+    truncate_ragged_lines = FALSE,
+    decimal_comma = FALSE,
+    glob = TRUE,
+    storage_options = NULL,
+    # credential_provider = NULL,
+    retries = 2,
+    file_cache_ttl = NULL,
+    include_file_paths = NULL) {
+  wrap({
+    check_dots_empty0(...)
+    check_character(source)
+    if (length(source) == 0) {
+      abort("`source` must have length > 0.")
+    }
+    check_list_of_polars_dtype(schema_overrides, allow_null = TRUE)
+    check_list_of_polars_dtype(schema, allow_null = TRUE)
+    check_named_list(schema_overrides, allow_null = TRUE)
+    check_named_list(schema, allow_null = TRUE)
+    check_named_list(storage_options, allow_null = TRUE)
+    encoding <- arg_match0(encoding, values = c("utf8", "utf8-lossy"))
+
+    if (isFALSE(infer_schema)) {
+      infer_schema_length <- 0
+    }
+
+    if (is.list(null_values)) {
+      all_character <- all(vapply(null_values, is_character, n = 1, FUN.VALUE = logical(1)))
+      if (!all_character) {
+        abort("When `null_values` is a list, each element must be of type character.")
+      }
+    } else if (is.character(null_values)) {
+      null_values <- as.list(null_values)
+    } else if (!is.null(null_values)) {
+      abort("`null_values` must be a character vector or a named list.")
+    }
+
+    schema_overrides <- parse_into_list_of_datatypes(!!!schema_overrides)
+
+    if (length(schema) > 0) {
+      schema <- parse_into_list_of_datatypes(!!!schema)
+    }
+
+    if (is.null(row_index_name) && !is.null(row_index_offset)) {
+      row_index_offset <- NULL
+    }
+
+    if (!is.null(storage_options)) {
+      all_character <- all(vapply(storage_options, is_character, n = 1, FUN.VALUE = logical(1)))
+      if (!all_character) {
+        abort("All elements of `storage_options` must be of type character.")
+      }
+    }
```

Comment:
Ditto.
My understanding is that a named vector is sufficient for this.

Something like: https://github.com/eitsupi/r-glaredb/blob/d011a337187e42421c7db08b75515599ba7651be/src/rust/src/connect.rs#L52-L71

Reply:
Done